### ISSUE TITLE
[angular-cli-hooks] TypeScript exports do not work

### ISSUE THREAD

**Issue by fidian** [blidblid/berglund#4]
In order to use this library with a TypeScript-based hook module, I have come across the following issues:

1. Peer dependencies are difficult to work with in my hook module. I include your module because I need the `hook()` function, but that needs a bunch of peer dependencies - those really aren't needed in order to compile a hook module.
2. None of the TypeScript exports are really exported in the library. That means I have to basically copy them into my project so my module can be typed.
3. I can't update the Angular schema, so adding extra options causes the builder to fail validation of `angular.json` because additional properties are not allowed.
4. I need to type the calls to `hook()` because it's not automatically figuring out what the parameters should be. Sample of code is below.

```
const hooks = [
    hook({
        name: 'build',
        schema: {}, // Doesn't automatically get typed as JSONSchema7 without later line
        before: runAtomizer,
    } as Hook<'build', JSONSchema7>),  // This is the part I would like to skip
];

function runAtomizer(
    ignore: any,
    context: BuilderContext
): Promise<BuilderOutput> {
```

A full example project would be fantastic to help illustrate how to do this correctly. I'm willing to publish mine, but it doesn't work with dev-server. I would really like a build step to work before every time webpack does anything and get it working before I publish it.

**Comment by blidblid** [comment]
Thanks for reporting an issue!

I'll fix 2. and 4. by emitting type declarations from `@berglund/angular-cli-hooks`.

For issue 3, what version of Angular are you using? I know they checked additional properties in earlier version, but AFAIK they removed that in Angular 13.

**Comment by blidblid** [comment]
You can try running with version `0.0.6`. It includes type declarations.

**Comment by fidian** [comment]
2 and 4 look good.

Peer dependencies are still producing problems with a current version of `@angular-devkit/architect`. Maybe use `>=0.1100.0` or another version specifier that allows for intermediate releases?

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @drn/angular-atomizer-hooks@0.0.1
npm ERR! Found: @angular-devkit/architect@0.1302.5
npm ERR! node_modules/@angular-devkit/architect
npm ERR!   @angular-devkit/architect@"^0.1302.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular-devkit/architect@"~0.1100.0 || ~0.1200.0 || ~0.1300.0" from @berglund/angular-cli-hooks@0.0.6
npm ERR! node_modules/@berglund/angular-cli-hooks
npm ERR!   dev @berglund/angular-cli-hooks@"^0.0.6" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/takins5/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/takins5/.npm/_logs/2022-02-25T14_46_05_095Z-debug-0.log
```

I am using Angular 13. I included additional properties into the schema and to the builder options, but it still didn't work for me without errors. I think this might be a separate issue and will make a new one.

**Comment by fidian** [comment]
List item 3 addressed in #6

**Comment by blidblid** [comment]
>Maybe use >=0.1100.0 or another version specifier that allows for intermediate releases?

I published 0.0.7 with these changes.

**Comment by fidian** [comment]
Works for me, thank you.
